integration/proxy/teleterm_test.go

```diff
@@ -53,26 +53,36 @@ func testTeletermGatewaysCertRenewal(t *testing.T, pack *dbhelpers.DatabasePack)
 		databaseURI := uri.NewClusterURI(rootClusterName).
 			AppendDB(pack.Root.MysqlService.Name)
 
-		testGatewayCertRenewal(t, pack, creds, databaseURI)
+		testGatewayCertRenewal(t, pack, "", creds, databaseURI)
 	})
 	t.Run("leaf cluster", func(t *testing.T) {
 		leafClusterName := pack.Leaf.Cluster.Secrets.SiteName
 		databaseURI := uri.NewClusterURI(rootClusterName).
 			AppendLeafCluster(leafClusterName).
 			AppendDB(pack.Leaf.MysqlService.Name)
 
-		testGatewayCertRenewal(t, pack, creds, databaseURI)
+		testGatewayCertRenewal(t, pack, "", creds, databaseURI)
+	})
+	t.Run("ALPN connection upgrade", func(t *testing.T) {
+		// Make a mock ALB which points to the Teleport Proxy Service. Then
+		// ALPN local proxies will point to this ALB instead.
+		albProxy := mustStartMockALBProxy(t, pack.Root.Cluster.Web)
+
+		databaseURI := uri.NewClusterURI(rootClusterName).
+			AppendDB(pack.Root.MysqlService.Name)
+
+		testGatewayCertRenewal(t, pack, albProxy.Addr().String(), creds, databaseURI)
 	})
 }
 
-func testGatewayCertRenewal(t *testing.T, pack *dbhelpers.DatabasePack, creds *helpers.UserCreds, databaseURI uri.ResourceURI) {
+func testGatewayCertRenewal(t *testing.T, pack *dbhelpers.DatabasePack, albAddr string, creds *helpers.UserCreds, databaseURI uri.ResourceURI) {
 	tc, err := pack.Root.Cluster.NewClientWithCreds(helpers.ClientConfig{
 		Login:   pack.Root.User.GetName(),
 		Cluster: pack.Root.Cluster.Secrets.SiteName,
+		ALBAddr: albAddr,
 	}, *creds)
 	require.NoError(t, err)
 	// The profile on disk created by NewClientWithCreds doesn't have WebProxyAddr set.
```

Comment:
I was wondering why we can now remove the `tc.WebProxyAddr` but not `tc.SaveProfile`. After some digging, I found out that the comment is now a bit outdated. We have to call `tc.SaveProfile` because `NewClientWithCreds` doesn't save the profile file to disk.

```suggestion
	// Save the profile yaml file to disk as NewClientWithCreds doesn't do that by itself.
```

Could you also update it here?

https://github.com/gravitational/teleport/blob/552af7f60a43e1eb1fe4cd7b051fbcfb492b9053/integration/teleterm_test.go#L225-L227